airflow/providers/amazon/aws/operators/glue.py

```diff
@@ -0,0 +1,117 @@
+    :type s3_bucket: Optional[str]
+    :param iam_role_name: AWS IAM Role for Glue Job Execution
+    :type iam_role_name: Optional[str]
+    """
+    template_fields = ()
+    template_ext = ()
+    ui_color = '#ededed'
+
+    @apply_defaults
+    def __init__(self,
+                 job_name='aws_glue_default_job',
+                 job_desc='AWS Glue Job with Airflow',
+                 script_location=None,
+                 concurrent_run_limit=None,
+                 script_args=None,
+                 connections=None,
+                 retry_limit=None,
+                 num_of_dpus=6,
+                 aws_conn_id='aws_default',
+                 region_name=None,
+                 s3_bucket=None,
+                 iam_role_name=None,
+                 *args, **kwargs
+                 ):
+        super(AWSGlueJobOperator, self).__init__(*args, **kwargs)
+        self.job_name = job_name
+        self.job_desc = job_desc
+        self.script_location = script_location
+        self.concurrent_run_limit = concurrent_run_limit
+        self.script_args = script_args or {}
+        self.connections = connections or []
+        self.retry_limit = retry_limit
+        self.num_of_dpus = num_of_dpus
+        self.aws_conn_id = aws_conn_id,
+        self.region_name = region_name
+        self.s3_bucket = s3_bucket
+        self.iam_role_name = iam_role_name
+        self.S3_PROTOCOL = "s3://"
+        self.S3_ARTIFACTS_PREFIX = 'artifacts/glue-scripts/'
+
+    def execute(self, context):
+        """
+        Executes AWS Glue Job from Airflow
+
+        :return: the id of the current glue job.
+        """
+        if not self.script_location.startsWith(self.S3_PROTOCOL):
+            s3_hook = S3Hook(aws_conn_id=self.aws_conn_id)
+            script_name = os.path.basename(self.script_location)
+            s3_hook.load_file(self.script_location, self.s3_bucket, self.S3_ARTIFACTS_PREFIX + script_name)
+        glue_job = AwsGlueJobHook(job_name=self.job_name,
+                                  desc=self.job_desc,
+                                  concurrent_run_limit=self.concurrent_run_limit,
+                                  script_location=self.script_location,
+                                  conns=self.connections,
+                                  retry_limit=self.retry_limit,
+                                  num_of_dpus=self.num_of_dpus,
+                                  aws_conn_id=self.aws_conn_id,
+                                  region_name=self.region_name,
+                                  s3_bucket=self.s3_bucket,
+                                  iam_role_name=self.iam_role_name)
+        self.log.info("Initializing AWS Glue Job: %s",self.job_name)
+        glue_job_run = glue_job.initialize_job(self.script_args)
+        self.log.info("AWS Glue Job: %s status: %s. Run Id: %s",self.job_name,glue_job_run['JobRunState'],glue_job_run['JobRunId'])
+        return glue_job_run['JobRunId']
```

Comment:
If `glue_job_run['JobRunState']` is `FAILED`, the operator still return 0 as a successful task in the dag run. We need to check the state and raise an `AirflowException` if job fails.

Reply:
I'm working on a local glue implementation as I can't wait until Airflow is available.  I have solved this with

```success_states = ['SUCCEEDED']
            errored_states = ['FAILED', 'STOPPED', 'TIMEOUT']

            if job_run_state in success_states:
                self.log.info("Exiting Job %s Run State: %s", run_id, job_run_state)
                return {'JobRunState': job_run_state, 'JobRunId': run_id}
            elif job_run_state in errored_states:
                job_error_message = "Exiting Job " + run_id + " Run State: " + job_run_state
                #self.log.info(job_error_message)
                raise AirflowException(job_error_message)
            else:
                self.log.info("Polling for AWS Glue Job %s current run state with status %s"
                    , job_name, job_run_state)
                time.sleep(self.JOB_POLL_INTERVAL)
```